builtin/plugin/v5/backend.go

```diff
@@ -0,0 +1,145 @@
+package plugin
+
+import (
+	"context"
+	"net/rpc"
+	"sync"
+
+	uuid "github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/logical"
+	bplugin "github.com/hashicorp/vault/sdk/plugin"
+	v5 "github.com/hashicorp/vault/sdk/plugin/v5"
+)
+
+// Backend returns an instance of the backend, either as a plugin if external
+// or as a concrete implementation if builtin, casted as logical.Backend.
+func Backend(ctx context.Context, conf *logical.BackendConfig) (logical.Backend, error) {
+	var b backend
+	name := conf.Config["plugin_name"]
+	pluginType, err := consts.ParsePluginType(conf.Config["plugin_type"])
+	if err != nil {
+		return nil, err
+	}
+
+	sys := conf.System
+
+	raw, err := v5.NewBackend(ctx, name, pluginType, sys, conf)
+	if err != nil {
+		return nil, err
+	}
+	b.Backend = raw
+	b.config = conf
+
+	return &b, nil
+}
+
+// backend is a thin wrapper around plugin.BackendPluginClient
+type backend struct {
+	logical.Backend
+	sync.RWMutex
```

Comment:
Similar reasoning here. If I named the logical.Backend here it would require writing forwarding methods. This is a convenience so I can just pass this along as a logical.Backend.